File: CS/AuthenticationOwin.Module/IAuthenticationOAuthUser.cs
using AuthenticationOwin.Module.BusinessObjects;
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AuthenticationOwin.Module {
    public interface IAuthenticationOAuthUser {
        string UserName { get; set; }
        XPCollection<EmailEntity> OAuthAuthenticationEmails { get; }
        bool EnableStandardAuthentication { get; }
    }
}

File: VB/AuthenticationOwin.Module/IAuthenticationOAuthUser.vb
Imports AuthenticationOwin.Module.BusinessObjects
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace AuthenticationOwin.Module
    Public Interface IAuthenticationOAuthUser
        Property UserName() As String
        ReadOnly Property OAuthAuthenticationEmails() As XPCollection(Of EmailEntity)
        ReadOnly Property EnableStandardAuthentication() As Boolean
    End Interface
End Namespace
